tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -1013,6 +1014,37 @@ void Matmul::validate(
                     }
                 }
                 TT_FATAL(input_tensor_b.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED);
+            } else if constexpr (std::is_same_v<
+                                     ProgramConfigType,
+                                     MatmulMultiCoreReuseMultiCastDRAMShardedProgramConfig>) {
+                TT_FATAL(input_tensor_a.is_sharded());
+                TT_FATAL(this->output_mem_config.is_sharded());
+                TT_FATAL(program_config.fuse_batch);
+                TT_FATAL(input_tensor_a.memory_config().memory_layout == TensorMemoryLayout::WIDTH_SHARDED);
+                TT_FATAL(input_tensor_a.memory_config().buffer_type == this->output_mem_config.buffer_type);
+                TT_FATAL(input_tensor_a.memory_config().memory_layout == this->output_mem_config.memory_layout);
+                TT_FATAL(input_tensor_a.shard_spec().value().orientation == ShardOrientation::ROW_MAJOR);
+                uint32_t M =
+                    (program_config.fuse_batch ? input_tensor_a.volume() / input_tensor_a.get_legacy_shape()[-1]
+                                               : input_tensor_a.get_legacy_shape()[-2]) /
+                    TILE_HEIGHT;
+                uint32_t N = input_tensor_b.get_legacy_shape()[-1] / TILE_WIDTH;
+                uint32_t K = input_tensor_a.get_legacy_shape()[-1] / TILE_WIDTH;
+                uint32_t per_core_M = program_config.per_core_M;
+                uint32_t per_core_N = program_config.per_core_N;
+                auto shard_shape = input_tensor_a.shard_spec().value().shape;
+
+                // No padding
+                TT_FATAL(M == per_core_M);
+                TT_FATAL(per_core_M == (shard_shape[0] / TILE_HEIGHT));
+                TT_FATAL(K % program_config.in0_block_w == 0);
+                TT_FATAL((shard_shape[1] / TILE_WIDTH) % program_config.in0_block_w == 0);
+                // TT_FATAL(div_up(N, per_core_N) == input_tensor_a.shard_spec().value().grid.num_cores());
```

Comment:
remove

Reply:
why remove the checks? is it checked somewhere else?